example-playbooks/replace_node/replace_node.yml

```diff
@@ -1,64 +1,74 @@
+# the cluster by blocking its ip via iptables.
+- name: Prepare nodes before replacement
+  hosts: scylla
   tasks:
-    - name: check for the replaced node being a seed
-      fail:
-        msg: "The node being replaced {{ replaced_node }} is a seed node. Please demote it to a non-seed first and try again."
-      when: replaced_node in scylla_seeds
+    - name: Resolve a scylla_listen_address as a fact
+      set_fact:
+        listen_address: "{{ alive_nodes_listen_address }}"
 
-  roles:
-    - ansible-scylla-node
+    - name: Resolve scylla_broadcast_address as a fact
+      set_fact:
+        broadcast_address: "{{ alive_nodes_broadcast_address }}"
 
+    - fail:
+        msg: "The replaced node can't be a seed!"
+      when: replaced_node_broadcast_address in scylla_seeds
+      run_once: true
 
-- name: Start scylla services
-  # new_node should be the IP of the node that comes into the cluster. The node being replaced should be in `replaced_node` variable below
-  hosts: new_node
-  become: true
+    - fail:
+        msg: "The new node can't be a seed!"
+      when: new_node in scylla_seeds
+      run_once: true
 
-  tasks:
-    - name: start scylla
-      service:
-        name: scylla-server
-        state: started
-      become: true
-      ignore_errors: true
-      #TODO: remove ignore_errors when ansible is bumped to 2.10.4 or 2.9.16 as per https://github.com/ansible/ansible/issues/71528
+    - name: Save one of the seeds as a fact in order to run some checks
+      set_fact:
+        alive_node: "{{ item }}"
+      when: item != replaced_node and item != new_node
+      loop: "{{ scylla_seeds }}"
+      run_once: true
 
-    - name: wait for the cluster to become healthy
-      shell: |
-        nodetool status|grep -E '^UN|^UJ|^DN'|grep "{{ vars['ansible_'~scylla_nic].ipv4.address }}"| wc -l
-      register: node_count
-      until: node_count.stdout|int == 1
-      retries: 300
-      delay: 1
+    - name: Check if CQL port is up on {{ replaced_node_broadcast_address }}
+      wait_for:
+        port: "{{ cql_port }}"
+        host: "{{ replaced_node_broadcast_address }}"
+        timeout: 60
+      register: _wait_for_cql_port_output
+      ignore_errors: true
+      delegate_to: "{{ alive_node }}"
+      run_once: true
 
-    - name: remove the replace_address_first_boot record
-      lineinfile:
-        path: /etc/scylla/scylla.yaml
-        regexp: '^replace_address_first_boot:'
-        state: absent
-      become: true
+    - name: Validate that the node being replaced is down
+      fail:
+        msg: "The node {{ replaced_node }} must be down in order to be replaced!"
+      when: _wait_for_cql_port_output.failed|bool == false
+      run_once: true
 
-    - name: run nodetool repair -pr
-      shell: |
-        nodetool repair -pr
+    - name: Check if the other nodes are UP
+      wait_for:
+        port: "{{ cql_port }}"
+        host: "{{ listen_address }}"
```

Comment:
Shouldn't you be checking `broadcast_rpc_address`?
For example, on AWS with a public IP configured and used for CQL `listen_address` is going to be an internal IP while clients (and this playbook) should connect to a CQL port via the public IP. 
And this is configured via `broadcast_rpc_address` parameter.

Reply:
These tasks run in the nodes themselves. So for node A this task will be executed in node A, and for node B it'll be executed in node B, so I think it makes sense to use `listen_address` here, right?